### ISSUE TITLE
records: provide a way of importing multiple records into a local instance

### ISSUE THREAD

**Issue by GraemeWatt** [HEPData/hepdata#275]
Currently, developers need to populate a local instance by manually downloading YAML `.tar.gz` files from https://hepdata.net and uploading each one via their local web interface.  The YAML `.tar.gz` files do not include possible auxiliary files linked from `additional_resources`.  Alternatively, they can run `hepdata utils load` to download from the old HepData site (http://hepdata.cedar.ac.uk), but it does not contain recent records after 2017.

It would be useful to have a CLI command for developers to download finalised records from hepdata.net as a `.zip` archive including both YAML and auxiliary files (see #110), then automatically populate their local instance.  This would be analogous to `hepdata utils load` but downloading from https://hepdata.net rather than http://hepdata.cedar.ac.uk.

**Comment by alisonrclarke** [comment]
Work in progress is on import-from-hepdata-net branch. Putting it on hold until #290 is complete.

**Comment by alisonrclarke** [comment]
@GraemeWatt Does this look OK as a set of commands? Let me know if you think any other options would be useful.

```bash
$ hepdata importer
Usage: hepdata importer [OPTIONS] COMMAND [ARGS]...

  Import from HepData website to a local instance.

Options:
  --help  Show this message and exit.

Commands:
  bulk-import-records  Populate the DB with records from HEPData.net (or...
  import-records       Populate the DB with specific records from...

$ hepdata importer bulk-import-records --help
Usage: hepdata importer bulk-import-records [OPTIONS]

  Populate the DB with records from HEPData.net (or another instance as
  specified by base_url)

  Usage: ``hepdata importer bulk-import-records -u -d 2020-01-01``

Options:
  -d, --date [%Y-%m-%d]          Filter all records modified since some point
                                 in time, e.g. 2016-07-05 for the 5th July
                                 2016.

  -u, --update-existing BOOLEAN  Whether to update records which already
                                 exist(defaults to False)

  -b, --base-url TEXT            Base URL from which to get data (defaults to
                                 https://hepdata.net)

  -n, --n-latest INTEGER         Get only the n most recently updated records
  --help                         Show this message and exit.

$ hepdata importer import-records --help
Usage: hepdata importer import-records [OPTIONS]

  Populate the DB with specific records from HEPData.net (or another
  instance as specified by base_url)

  Usage: ``hepdata importer import-records -i 'ins1262703' -rc False``

Options:
  -i, --inspireids TEXT          A comma separated list of recids to load.
  -rc, --recreate_index BOOLEAN  Whether or not to recreate the index before
                                 importing(defaults to False)

  -u, --update-existing BOOLEAN  Whether to update records which already
                                 exist(defaults to False)

  -b, --base-url TEXT            Base URL from which to get data (defaults to
                                 https://hepdata.net)

  --help                         Show this message and exit.
```

**Comment by GraemeWatt** [comment]
Looks ideal, thanks!  Minor suggestion: maybe another option `-r, --recids TEXT` for `import-records` to use the original HEPData record IDs rather than the INSPIRE record IDs?  The HEPData record IDs will not be preserved in the new instance, but this might still be a useful option.

**Comment by alisonrclarke** [comment]
Hmmm...actually looking at the way the import works, adding a `--recids` option would be less straightforward than I first thought. We'd need to download and process the file before we can check whether it's a submission that already exists locally. It's do-able, but I just wonder under what circumstances it would actually be used, and therefore whether it's worth the effort, as this is only something that developers will use?

**Comment by GraemeWatt** [comment]
OK, don't worry about it if it's not easy to implement.  I only suggested it because most other CLI commands take either a HEPData ID or an INSPIRE ID as arguments.  The existing options are fine.
